src/main/java/core/basesyntax/model/FruitTransaction.java

```diff
@@ -0,0 +1,50 @@
+package core.basesyntax.model;
+
+import java.util.Arrays;
+
+public class FruitTransaction {
+    public enum Operation {
+        BALANCE("b"), SUPPLY("s"), PURCHASE("p"), RETURN("r");
+
+        private final String code;
+
+        Operation(String code) {
+            this.code = code;
+        }
+
+        public String getCode() {
+            return code;
+        }
+
+        public static Operation fromCode(String code) {
+            return Arrays.stream(values())
+                    .filter(op -> op.getCode().equals(code))
+                    .findFirst()
+                    .orElseThrow(() -> new IllegalArgumentException(
+                            "Invalid operation code: " + code)
+                    );
+        }
+    }
+
+    private final Operation operation;
+    private final String fruit;
+    private final int quantity;
+
+    public FruitTransaction(Operation operation, String fruit, int quantity) {
+        this.operation = operation;
+        this.fruit = fruit;
+        this.quantity = quantity;
+    }
+
+    public Operation getOperation() {
+        return operation;
+    }
+
+    public String getFruit() {
+        return fruit;
+    }
+
+    public int getQuantity() {
+        return quantity;
+    }
+}
```

Comment:
The class is missing a `toString` method, which is often useful for logging and debugging purposes. Consider adding a `toString` method that returns a string representation of the `FruitTransaction` object.